extensions/ql-vscode/src/model-editor/shared/sorting.ts

```diff
@@ -27,12 +28,76 @@ export function sortGroupNames(
   );
 }
 
-export function sortMethods(methods: readonly Method[]): Method[] {
+/**
+ * Primarily sorts methods into the following order:
+ * - Unsaved positive AutoModel predictions
+ * - Negative AutoModel predictions
+ * - Unsaved manual models
+ * - Umodeled
+ * - Modeled and saved (AutoModel and manual)
+ *
+ * Secondary sort order is by number of usages descending, then by method signature ascending.
+ */
+export function sortMethods(
+  methods: readonly Method[],
+  modeledMethodsMap: Record<string, readonly ModeledMethod[]>,
+  modifiedSignatures: ReadonlySet<string>,
+  processedByAutoModelMethods: ReadonlySet<string>,
+): Method[] {
   const sortedMethods = [...methods];
-  sortedMethods.sort((a, b) => compareMethod(a, b));
+  sortedMethods.sort((a, b) => {
+    const methodAPrimarySortOrdinal = getMethodPrimarySortOrdinal(
+      !!modeledMethodsMap[a.signature]?.length,
+      modifiedSignatures.has(a.signature),
+      processedByAutoModelMethods.has(a.signature),
+    );
+    const methodBPrimarySortOrdinal = getMethodPrimarySortOrdinal(
+      !!modeledMethodsMap[b.signature]?.length,
+      modifiedSignatures.has(b.signature),
+      processedByAutoModelMethods.has(b.signature),
+    );
+    if (methodAPrimarySortOrdinal !== methodBPrimarySortOrdinal) {
+      return methodAPrimarySortOrdinal - methodBPrimarySortOrdinal;
+    }
+
+    // Then sort by number of usages descending
+    const usageDifference = b.usages.length - a.usages.length;
+    if (usageDifference !== 0) {
+      return usageDifference;
+    }
+
+    // Then sort by method signature ascending
+    return a.signature.localeCompare(b.signature);
+  });
   return sortedMethods;
 }
 
+/**
+ * Assigns numbers to the following classes of methods:
+ * - Unsaved positive AutoModel predictions => 0
+ * - Negative AutoModel predictions => 1
+ * - Unsaved manual models => 2
+ * - Umodeled => 3
+ * - Modeled and saved (AutoModel and manual) => 4
+ */
```

Comment:
Yes, you're right. We should make sure to put methods that are modeled in other model packs at the bottom.

Trying it out I've also realised another change I want to make. We be sorting "unmodeled methods" and "unsaved manual models" together, because otherwise when you make a manual modeling the row instantly disappears as it gets sorted differently.